### PR TITLE
Dependency update

### PR DESCRIPTION
**Pull request by daniel-noland**
I bumped the dependencies to current everywhere I could find something out of date.

I didn't notice any regressions in this patch series although some minor refactoring was required.

I left the series as distinct, easily reverted patches in case something goes wrong.  Let me know if you want me to squash them.

### CODE REVIEW

File: docker-compose.yml [hunk 1]
```diff
@@ -2,10 +2,10 @@
...
 services:
   app:
-    image: jasonwhite0/rudolfs:latest
-    #build:
-    #  context: .
-    #  dockerfile: Dockerfile
+    image: rudolfs:local
+    build:
+     context: .
+     dockerfile: Dockerfile
```
Comment (jasonwhite) [inline docker-compose.yml:8]: Did you intend to commit these changes? Looks like this was just changed for local development.

**Review by jasonwhite — CHANGES_REQUESTED** [review]
Awesome! Thanks for this. Just one minor question before I merge it.

File: src/storage/cached.rs [hunk 2]
```diff
@@ -29,7 +29,7 @@ use futures::{
     future::{self, FutureExt, TryFutureExt},
     stream::{StreamExt, TryStreamExt},
 };
-use humansize::{file_size_opts as file_size, FileSize};
+use humansize;
```
Comment (jasonwhite) [inline src/storage/cached.rs:32]: This line can be removed.

**Review by jasonwhite — CHANGES_REQUESTED** [review]

**Comment by jasonwhite** [comment]
FYI, I fixed the clippy lint errors in the main branch.

File: docker-compose.yml (hunk 1 above)
Comment (daniel-noland) [reply docker-compose.yml:8]: Oh, nope!

That is me being sloppy with version control.

I'll fix it.

File: src/storage/cached.rs (hunk 2 above)
Comment (daniel-noland) [reply src/storage/cached.rs:32]: true.  Will fix it

File: docker-compose.yml (hunk 1 above)
Comment (daniel-noland) [reply docker-compose.yml]: this should be sorted now

File: src/storage/cached.rs (hunk 2 above)
Comment (daniel-noland) [reply src/storage/cached.rs]: fixed

**Comment by daniel-noland** [comment]
> FYI, I fixed the clippy lint errors in the main branch.

I rebased onto `master` so clippy warnings should not bounce me in CI this time.

Thanks for sorting them :)

**Comment by daniel-noland** [comment]
one note vs prior version of this PR:

I didn't even see the `faulty` feature when I did the prior batch of updates.

As a result my changes introduced several regressions I didn't catch on that code base.

I cleaned them up in the second to final commit in this series.  That commit is worth review at your convenience.

If you want me to move that to a different patch series that's fine.

Alternatively I can re-structure the patch series to make sure `faulty` doesn't regress as we walk the updates.  As this series stands there are commits in the tree where `faulty` won't build

**Pull request merged**